src/server/routes/oauth.ts

```diff
@@ -0,0 +1,109 @@
+import { Request, Router } from 'express';
+import { ResponseWithRequestState } from '@/server/models/Express';
+import { Routes } from '@/shared/model/Routes';
+import {
+  deleteAuthorizationStateCookie,
+  getAuthorizationStateCookie,
+  OIDCProfileClient,
+} from '@/server/lib/okta/oidc';
+import { getConfiguration } from '@/server/lib/getConfiguration';
+import { logger } from '@/server/lib/logger';
+import { trackMetric } from '@/server/lib/trackMetric';
+import { Metrics } from '@/server/models/Metrics';
+import { handleAsyncErrors } from '@/server/lib/expressWrappers';
+import { exchangeAccessTokenForCookies } from '@/server/lib/idapi/auth';
+import { setIDAPICookies } from '@/server/lib/setIDAPICookies';
+import { stringify } from 'querystring';
+import { SignInErrors } from '@/shared/model/Errors';
+
+const router = Router();
+
+const { signInPageUrl, defaultReturnUri } = getConfiguration();
+
+router.get(
+  Routes.OAUTH_AUTH_CODE_CALLBACK,
+  handleAsyncErrors(async (req: Request, res: ResponseWithRequestState) => {
+    try {
+      // params returned from the /authorize endpoint
+      // for auth code flow they will be "code" and "state"
+      // "code" is the authorization code to exchange for access token
+      // "state" will be the "nonce" value set in the oidc_auth_state cookie
+      // if there were any errors, then an "error", and "error_description" params
+      // will be returned instead
+      const callbackParams = OIDCProfileClient.callbackParams(req);
+
+      // get the oidc_auth_state cookie which contains the "nonce"
+      // and "returnUrl" so we can get the user back to the page they
+      // initially landed on sign in from
+      const authState = getAuthorizationStateCookie(req);
+
+      // check if the state cookie exists
+      if (!authState) {
+        // no state! is this an attack?
+        // redirect to sign in and cancel flow
+        trackMetric(Metrics.AUTHORIZATION_FAILURE);
+        return res.redirect(signInPageUrl);
+      }
+
+      // we have the Authorization State now, so the cookie is
+      // no longer required, so mark cookie for deletion in the response
+      deleteAuthorizationStateCookie(res);
+
+      // exchange the auth code for access token + id token
+      // and check the "state" we got back from the callback
+      // to the "nonce" that was set in the AuthorizationState
+      const tokenSet = await OIDCProfileClient.oauthCallback(
+        // the redirectUri is the callback location (this route)
+        OIDCProfileClient.metadata.redirect_uris?.[0],
```

Comment:
it was mentioned in the team review to refer to the constant rather than using `[0]`